pkg/core/hash.go

```diff
@@ -64,6 +64,38 @@ func calculateConfigHash(children []Object) (string, error) {
 	return fmt.Sprintf("%x", hashBytes), nil
 }
 
+// getConfigMapData extracts all the relevant data from the ConfigMap, whether that is
+// the whole ConfigMap or only the specified keys.
+func getConfigMapData(child configObject) map[string]string {
+	cm := corev1.ConfigMap(*child.object.(*corev1.ConfigMap))
+	if child.allKeys {
+		return cm.Data
+	}
+	keyData := make(map[string]string)
+	for key := range child.keys {
+		if value, exists := cm.Data[key]; exists {
+			keyData[key] = value
+		}
+	}
+	return keyData
+}
+
+// getSecretData extracts all the relevant data from the Secret, whether that is
+// the whole Secret or only the specified keys.
+func getSecretData(child configObject) map[string][]byte {
+	s := corev1.Secret(*child.object.(*corev1.Secret))
```

Comment:
Don't need the comment after the type assertion
```suggestion
	s := *child.object.(*corev1.Secret)
```

Reply:
Fixed